rackcity/views/change_plan_views.py

```diff
@@ -0,0 +1,294 @@
+    """
+    List many change plans. If page is not specified as a query parameter, all
+    models are returned. If page is specified as a query parameter, page
+    size must also be specified, and a page of models will be returned.
+    """
+    errors = []
+    should_paginate = not(
+        request.query_params.get('page') is None
+        and request.query_params.get('page_size') is None
+    )
+    if should_paginate:
+        if not request.query_params.get('page'):
+            errors.append("Must specify field 'page' on paginated requests")
+        elif not request.query_params.get('page_size'):
+            errors.append(
+                "Must specify field 'page_size' on paginated requests"
+            )
+        elif int(request.query_params.get('page_size')) <= 0:
+            errors.append(
+                "Field 'page_size' must be an integer greater than 0"
+            )
+
+    if len(errors) > 0:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.ERROR.value + GenericFailure.PAGE_ERROR.value,
+                "errors": " ".join(errors)
+            },
+            status=HTTPStatus.BAD_REQUEST,
+        )
+
+    user = request.user
+    change_plans_query = ChangePlan.objects.filter(owner=user)
+
+    try:
+        filter_args = get_filter_arguments(request.data)
+    except Exception as error:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.ERROR.value + GenericFailure.FILTER.value,
+                "errors": str(error)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    for filter_arg in filter_args:
+        change_plans_query = change_plans_query.filter(**filter_arg)
+    try:
+        sort_args = get_sort_arguments(request.data)
+    except Exception as error:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.ERROR.value + GenericFailure.SORT.value,
+                "errors": str(error)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    change_plans = change_plans_query.order_by(*sort_args)
+
+    if should_paginate:
+        paginator = PageNumberPagination()
+        paginator.page_size = request.query_params.get('page_size')
+        try:
+            page_of_change_plans = paginator.paginate_queryset(
+                change_plans, request)
+        except Exception as error:
+            return JsonResponse(
+                {
+                    "failure_message":
+                        Status.ERROR.value + GenericFailure.PAGE_ERROR.value,
+                    "errors": str(error)
+                },
+                status=HTTPStatus.BAD_REQUEST,
+            )
+        change_plans_to_serialize = page_of_change_plans
+    else:
+        change_plans_to_serialize = change_plans
+    serializer = GetChangePlanSerializer(change_plans_to_serialize, many=True)
```

Comment:
in response to my above comment, it seems like the 2 serializers are functionally identical, so I would collapse them into one (ChangePlanSerializer)